spec/README.md

```diff
@@ -0,0 +1,39 @@
+# Specification
+
+The specification aims to keep uniformity accross the libraries and to provide a human digestible output while producing a structured format.
+
+The ordering of the next three keys must be respected in every ecs-logging library:
+
+1. `@timestamp`, base field
+2. `log.level`, log field
+3. `message`, base field
+
+With the fourth key, `ecs.version` ([core](https://www.elastic.co/guide/en/ecs/current/ecs-ecs.html) field) in the [ND-JSON](https://github.com/ndjson/ndjson-spec) output, we define the *minimum viable product* (MVP) for a log line.
+`ecs.version` must be present in case of appenders that are not adding the `ecs.version` automatically.
+
+All other keys are not subjected to an order until decided differently and can hence be appended to the log line.
+
+## Examples
+
+### Minimum Viable Product
```

Comment:
I like the idea of listing a few "levels" of viability.

I like the current MVP level of `@timestamp`, log.level, message & ecs.version. It should be possible to deliver that in most languages with minimal effort.

I think we should also describe a more fleshed out level, like the Java library lists [here](https://github.com/elastic/ecs-logging-java#mapping). I see this level as still mostly "drop-in" + perhaps configuration around the expensive attributes to collect. But essentially, developers are still logging like they used to, without needing to "care" about ECS field definitions.

We've discussed eventually supporting the possibility to let developers fill more ECS fields if they care. That could be one more level of maturity of a library. None of them is ready to do so yet (afaik), but we can still define this as possible level of maturity we will be working towards.